lib/Target/TTNN/TTNNToFlatbuffer.cpp

```diff
@@ -721,22 +721,55 @@ createOp(FlatbufferObjectCache &cache, AllGatherOp op) {
   auto input = cache.at<::tt::target::ttnn::TensorRef>(
       getOperandThroughDPSOps(op.getInput()));
   auto output = cache.getOrCreate(op.getResult(), tensorValueToFlatbuffer);
-  auto device = getOperandThroughDPSOps(op.getDevice());
+  std::optional<uint8_t> subDeviceId =
+      op.getSubDeviceId() ? std::make_optional<uint8_t>(static_cast<uint8_t>(
+                                op.getSubDeviceId().value()))
+                          : std::nullopt;
+
+  // Convert optional values to flatbuffer format
+  auto memoryConfig = op.getMemoryConfig()
+                          ? toFlatbuffer(cache, op.getMemoryConfig().value())
+                          : 0;
+  auto numLinks = op.getNumLinks()
+                      ? std::make_optional<uint32_t>(op.getNumLinks().value())
+                      : std::nullopt;
+  auto topology = op.getTopology()
+                      ? std::make_optional<::tt::target::common::Topology>(
+                            static_cast<::tt::target::common::Topology>(
```

Comment:
:warning: **clang\-diagnostic\-error** :warning:
no type named `` Topology `` in namespace `` tt::target::common ``; did you mean `` ::tt::target::Topology ``?
```suggestion
                            static_cast<::tt::target::Topology>(
```